SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/EditSneakerEntryActivity.java

```diff
@@ -0,0 +1,178 @@
+import com.cyscorpions.dalejulian.sneakpeek.R.id;
+import com.cyscorpions.dalejulian.sneakpeek.R.layout;
+import com.cyscorpions.dalejulian.sneakpeek.models.Sneaker;
+import com.cyscorpions.dalejulian.sneakpeek.models.SneakerCategory;
+import com.cyscorpions.dalejulian.sneakpeek.models.SneakerDirectory;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+
+public class EditSneakerEntryActivity extends Activity {
+
+	private EditText mCategory;
+	private Button mSaveButton;
+	private EditText mName, mBrand, mRarity, mSellVal, mDesc;
+	private Sneaker mSneaker = new Sneaker();
+	private ImageView mThumbnail;
+
+	public static final String EDIT_NAME_EXTRA = "name";
+	public static final String EDIT_BRAND_EXTRA = "brand";
+	public static final String EDIT_SELLVAL_EXTRA = "sellval";
+	public static final String EDIT_CATEGORY_EXTRA = "categ";
+	public static final String EDIT_DESC_EXTRA = "desc";
+	public static final String EDIT_RARITY_EXTRA = "rare";
+
+	private String id;
+
+	@Override
+	public void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.sneaker_editentry_activity);
+
+		findResourceIds();
+		setupEditView();
+		getActionBar().setDisplayHomeAsUpEnabled(true);
+	}
+
+	public interface DataChangedListener {
+		void onDataChanged();
+	}
+
+	private void findResourceIds() {
+		mCategory = (EditText) findViewById(R.id.txtSneakerCategory);
+		mSaveButton = (Button) findViewById(R.id.btnSaveData);
+		mName = (EditText) findViewById(R.id.txtSneakerName);
+		mBrand = (EditText) findViewById(R.id.txtSneakerBrand);
+		mRarity = (EditText) findViewById(R.id.txtSneakerRarity);
+		mSellVal = (EditText) findViewById(R.id.txtSellingValue);
+		mDesc = (EditText) findViewById(R.id.txtSneakerDescription);
+		mThumbnail = (ImageView) findViewById(R.id.imgSneakerEntry);
+	}
+
+	private void saveSneaker() {
+		SneakerCategory category = new SneakerCategory();
+		category.setName(mCategory.getText().toString());
+		Sneaker sneaker = new Sneaker(mName.getText().toString(), mBrand
+				.getText().toString(), mRarity.getText().toString(), mSellVal
+				.getText().toString(), mDesc.getText().toString(), category, 0);
+
+		SneakerDirectory.get(this).addSneaker(sneaker);
+		Intent intent = new Intent();
+		setResult(RESULT_OK, intent);
+		finish();
+	}
+
+	private void setupEditView() {
+		boolean isFromDetailActivity = getIntent().getBooleanExtra(
+				SneakerDetailsActivity.DETAILS_FROMDETAILS_EXTRA, false);
+		boolean isFromListActivity = getIntent().getBooleanExtra(
+				SneakerEntryListActivity.LIST_FROMLIST_EXTRA, false);
+
+		if (isFromDetailActivity) {
+			mSneaker.setBrand(getIntent().getStringExtra(
```

Comment:
Do you really need to have different KEYS for your extras? They all have the same values. It would be better if you put the KEY definitions into this class instead and have the other classes statically get them from here. Also, the if-else conditions is not that useful since they do almost the same thing. You can put the extras passed into this class into Strings variables instead of creating a new Sneaker object.